docs/effects/simpleeff.rst

```diff
@@ -410,10 +410,28 @@ following type:
 
 This returns a dependent pair of a length, and a vector of that
 length, because we cannot know in advance how many integers the user
-is going to input. One way to implement this function, using ``-1`` to
-indicate the end of input, is shown in Listing [readvec]. This uses a
-variation on ``parseNumber`` which does not require a number to be
-within range.
+is going to input. We can use ``-1`` to indicate the end of input:
+
+.. code-block:: idris
+
+    read_vec : Eff (p ** Vect p Int) [STDIO]
+    read_vec = do putStr "Number (-1 when done): "
+                  case the (Maybe Int) $ run (parseNumber (trim !getStr)) of
```

Comment:
Is this really the nicest way to provide the missing type information, as opposed to (say) giving `run` an explicit type argument? And shouldn't the surrounding text also say something about _why_ we need to do this?